nautilus_core/execution/src/order_emulator/adapter.rs

```diff
@@ -0,0 +1,114 @@
+// -------------------------------------------------------------------------------------------------
+//  Copyright (C) 2015-2025 Nautech Systems Pty Ltd. All rights reserved.
+//  https://nautechsystems.io
+//
+//  Licensed under the GNU Lesser General Public License Version 3.0 (the "License");
+//  You may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at https://www.gnu.org/licenses/lgpl-3.0.en.html
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------------------------------
+
+use std::{
+    cell::{Ref, RefCell, RefMut},
+    rc::Rc,
+};
+
+use nautilus_common::{
+    cache::Cache,
+    clock::Clock,
+    msgbus::{handler::ShareableMessageHandler, MessageBus},
+};
+use nautilus_core::UUID4;
+use ustr::Ustr;
+
+use crate::{
+    messages::{
+        cancel::CancelOrderHandlerAny, modify::ModifyOrderHandlerAny, submit::SubmitOrderHandlerAny,
+    },
+    order_emulator::{
+        emulator::OrderEmulator,
+        handlers::{OrderEmulatorExecuteHandler, OrderEmulatorOnEventHandler},
+    },
+};
+
+pub struct OrderEmulatorAdapter {
+    emulator: Rc<RefCell<OrderEmulator>>,
+}
+
+impl OrderEmulatorAdapter {
+    pub fn new(
+        clock: Rc<RefCell<dyn Clock>>,
+        cache: Rc<RefCell<Cache>>,
+        msgbus: Rc<RefCell<MessageBus>>,
+    ) -> Self {
```

Comment:
A lot constructs are wrapped by Rc RefCell but it adds to the cognitive load. Perhaps all such constructs can defined a wrapper type alias to make it easier to read.

pub type SharedCache = Rc<RefCell<Cache>>;